src/snapshot/Serialize/SnapshotData.lua

```diff
@@ -0,0 +1,120 @@
+local RoactRoot = script.Parent.Parent.Parent
+local Markers = require(script.Parent.Markers)
+local ElementKind = require(RoactRoot.ElementKind)
+local Type = require(RoactRoot.Type)
+local Ref = require(RoactRoot.PropMarkers.Ref)
+
+local function sortSerializedChildren(childA, childB)
+	return childA.hostKey < childB.hostKey
+end
+
+local SnapshotData = {}
+
+function SnapshotData.type(wrapperType)
```

Comment:
I wonder if SnapshotData and Serializer could be the same thing - a single source of truth to go from Shallow to a string representing a ModuleScript (snapshot)

Reply:
I am not sure if we can do that because when we compare with the generated snapshot with the previous serialized one, we use that intermediate representation to do the comparison.

So I think we're kind of stuck with that intermediate representation! If you have any idea of how we could do it though let me know, because it would reduce one layer of complexity so that would be great

Reply:
Another alternative is to rename SnapshotData to Snapshot, and then Snapshot could be called SnapshotMatcher. It is nice to keep them separate from Serializer so you have a single entity to call deepEqual on as you have now